src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -16,7 +17,7 @@
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.CPS
 {
     [Export(typeof(IWorkspaceProjectContextFactory))]
-    internal partial class CPSProjectFactory : IWorkspaceProjectContextFactory
+    internal partial class CPSProjectFactory : ForegroundThreadAffinitizedObject, IWorkspaceProjectContextFactory
```

Comment:
Note this will also cause you to *assert* that the type is being created on the UI thread. You need to retest TypeScript, F# and XAML if you intend to take that path. Alternatively you can put a base constructor invocation to opt out.